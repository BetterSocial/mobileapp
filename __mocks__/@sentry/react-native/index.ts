export const addBreadcrumb = jest.fn();
