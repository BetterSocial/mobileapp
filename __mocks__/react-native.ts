import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN