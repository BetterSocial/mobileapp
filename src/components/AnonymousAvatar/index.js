import * as React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';

import AnonymousProfile from '../../assets/images/AnonymousProfile.png';
import dimen from '../../utils/dimen';
import {POST_VERSION} from '../../utils/constants';

export const styles = StyleSheet.create({
  imageAnonimity: {
    marginRight: 0,
    width: dimen.size.FEED_HEADER_IMAGE_RADIUS,
    height: dimen.size.FEED_HEADER_IMAGE_RADIUS
  },
  avatarV2Background: (backgroundColor, radius) => ({
    width: radius,
    height: radius,
    borderRadius: radius / 2,
    backgroundColor,
    justifyContent: 'center'
  }),
  avatarV2Emoji: (emojiRadius) => ({
    fontSize: emojiRadius,
    textAlign: 'center',
    alignSelf: 'center'
  })
});

/**
 * @typedef {Object} AnonymousAvatarComponentProps
 * @property {number} [radius=dimen.size.FEED_HEADER_IMAGE_RADIUS]
 * @property {number} [emojiRadius=dimen.size.FEED_HEADER_ANONYMOUS_IMAGE_RADIUS]
 * @property {string} version
 * @property {import('react-native').StyleProp} containerStyle
 * @property {AnonUserInfoTypes} anonUserInfo
 */

/**
 *
 * @param {AnonymousAvatarComponentProps} props
 */
const AnonymousAvatar = (props) => {
  const {
    version,
    anonUserInfo,
    containerStyle = {},
    radius = dimen.size.FEED_HEADER_IMAGE_RADIUS,
    emojiRadius = dimen.size.FEED_HEADER_ANONYMOUS_IMAGE_RADIUS
  } = props;

  if (version >= POST_VERSION) {
    return (
      <View
        testID="newVersion"
        style={{...styles.avatarV2Background(anonUserInfo.colorCode, radius), ...containerStyle}}>
        <Text style={styles.avatarV2Emoji(emojiRadius)}>{anonUserInfo.emojiCode}</Text>
      </View>
    );
  }

  return (
    <Image
      source={AnonymousProfile}
      width={radius}
      height={radius}
      style={{...styles.imageAnonimity, ...containerStyle}}
    />
  );
};

export default AnonymousAvatar;
