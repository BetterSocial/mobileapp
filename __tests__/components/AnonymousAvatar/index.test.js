import * as React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

import AnonymousAvatar from '../../../src/components/AnonymousAvatar';

describe('AnonymousAvatar test', () => {
  it('Should match version 1 snapshot', () => {
    const toJSON = renderer.create(<AnonymousAvatar version={1} radius={15} />).toJSON();
    expect(toJSON).toMatchSnapshot();
  });

  it('Should match version 2 snapshot', () => {
    const toJSON = renderer
      .create(
        <AnonymousAvatar
          version={2}
          anonUserInfo={{
            colorCode: '#000000',
            emojiCode: 'ðŸ˜€'
          }}
          radius={15}
          emojiRadius={10}
        />
      )
      .toJSON();
    expect(toJSON).toMatchSnapshot();
  });

  it('Should render the correct emoji', () => {
    const {queryByText} = render(
      <AnonymousAvatar
        version={2}
        anonUserInfo={{
          colorCode: '#000000',
          emojiCode: 'ðŸ˜€'
        }}
        radius={15}
        emojiRadius={10}
      />
    );

    expect(queryByText('ðŸ˜€')).toBeTruthy();
  });
});
