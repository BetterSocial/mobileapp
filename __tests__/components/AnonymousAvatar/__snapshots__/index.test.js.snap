// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnonymousAvatar test Should match version 1 snapshot 1`] = `
<Image
  height={15}
  source={
    Object {
      "testUri": "../../../src/assets/images/AnonymousProfile.png",
    }
  }
  style={
    Object {
      "height": NaN,
      "marginRight": 0,
      "width": NaN,
    }
  }
  width={15}
/>
`;

exports[`AnonymousAvatar test Should match version 2 snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "borderRadius": 7.5,
      "height": 15,
      "justifyContent": "center",
      "width": 15,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 10,
      }
    }
  >
    ðŸ˜€
  </Text>
</View>
`;
